### PR TITLE
Fix BigInt to Float conversions

### LINKED ISSUE #6365: convert(Float64, ::BigInt) – rounding and corner cases
See https://github.com/JuliaLang/julia/commit/f04aa6fc16177b01975a3af40c1747718e1db8c8 and https://gmplib.org/manual/Converting-Integers.html#Converting-Integers.

### PR DESCRIPTION
**Pull request by simonbyrne**
Fixes #6365

### CODE REVIEW

File: base/float.jl [hunk 1]
```diff
@@ -419,3 +419,7 @@ Rounding.set_rounding_raw(Float64, Rounding.JL_FE_TONEAREST)
 ## byte order swaps for arbitrary-endianness serialization/deserialization ##
 bswap(x::Float32) = box(Float32,bswap_int(unbox(Float32,x)))
 bswap(x::Float64) = box(Float64,bswap_int(unbox(Float64,x)))
+
+## checks if last bit is 0 or 1: used for IEEE rounding
+is_ieee_even(x::Float64) = reinterpret(UInt64,x) & 1 == 0
+is_ieee_even(x::Float32) = reinterpret(UInt32,x) & 1 == 0
```
Comment (StefanKarpinski) [inline base/float.jl]: Maybe write these as `is_ieee_even(x::Float64) = iseven(reinterpret(UInt64,x))`?

Comment (simonbyrne) [inline base/float.jl]: I have a slightly better solution.

Comment (StefanKarpinski) [inline base/float.jl]: Yes, generic reinterpret is much better!

**Comment by simonbyrne** [comment]
I'm not sure what's causing the appveyor breakage.

**Comment by tkelman** [comment]
It looks a little scarier than the usual case of #10045, but I just built this branch (just once for now) locally on win32 without seeing it. I'll restart the appveyor build.

**Comment by simonbyrne** [comment]
Bump. Any thoughts?

cc: @StefanKarpinski

**Comment by StefanKarpinski** [comment]
Seems like a good thing to fix. Is there some potential downside? Otherwise just do it.

**Comment by simonbyrne** [comment]
It will require some intermediate allocation of `BigInt`s, but only if the original argument is larger than 2^53. I can't really see any way around this.

**Pull request merged**